add support for css `@property` (#1092)
vanilla-extract-css/vanilla-extract: site/docs/api/create-var.md

## Assigning variables dynamically

CSS variables can also be assigned dynamically using APIs in [the `@vanilla-extract/dynamic` package](/documentation/packages/dynamic).

> ## CSS Properties
> 
> You can also create typed css [@property](https://developer.mozilla.org/en-US/docs/Web/CSS/@property) using `createVar`:
> 
> ```ts compiled
> // accent.css.ts
> 
> import { createVar, style } from '@vanilla-extract/css';
> 
> export const accentVar = createVar({
>   syntax: '<color>',
>   inherits: false,
>   initialValue: 'blue'
> });
> ```
> 
> CSS properties created with `createVar` can be used in the same way as a regular css variable:
> 
> ```ts compiled
> export const pink = style({
>   vars: {
>     [accentVar]: 'pink'
>   }
> });
> ```
> 
> The main benefit of using `createVar` to create typed css properties is that they can be [animated via keyframes](/documentation/api/keyframes#animating-variables).

Reviewer: ```suggestion
## @property rules

[@property] rules can also be created using `createVar`.
CSS variables with @property rules are used in the same way as regular CSS variables:

```ts compiled
// accent.css.ts
import { createVar, style } from '@vanilla-extract/css';

export const accentVar = createVar({
  syntax: '<color>',
  inherits: false,
  initialValue: 'blue'
});

export const pink = style({
  vars: {
    [accentVar]: 'pink'
  }
});
```

[@property]: https://developer.mozilla.org/en-US/docs/Web/CSS/@property
```

## Assigning variables dynamically

CSS variables can also be assigned dynamically using APIs in [the `@vanilla-extract/dynamic` package](/documentation/packages/dynamic).

## CSS Properties

You can also create typed css [@property](https://developer.mozilla.org/en-US/docs/Web/CSS/@property) using `createVar`:

```ts compiled
// accent.css.ts

import { createVar, style } from '@vanilla-extract/css';

export const accentVar = createVar({
  syntax: '<color>',
  inherits: false,
  initialValue: 'blue'
});
```

CSS properties created with `createVar` can be used in the same way as a regular css variable:

```ts compiled
export const pink = style({
  vars: {
    [accentVar]: 'pink'
  }
});
``` 

The main benefit of using `createVar` to create typed css properties is that they can be animated via `keyframes`.

```ts compiled
const angle = createVar({
  syntax: '<angle>',
  inherits: false,
  initialValue: '0deg',
});

const angleKeyframes = keyframes({
  '0%': {
    [getVarName(angle)]: '0deg',
  },
  '100%': {
    [getVarName(angle)]: '360deg',
  },

> });

Author: hey @askoufis, thanks for your feedback, this is a really good improvement!

    [accentVar]: 'pink'
  }
});

> ``` 

Author: moved to `keyframes` doc
